### PR TITLE
fixes #7230 - rails 4 compatibility

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
related to https://github.com/theforeman/foreman/pull/2870

### CODE REVIEW

**Comment by johnpmitsch** [comment]
[test]

**Comment by unorthodoxgeek** [comment]
not sure that these code changes are backward-compatible, tests might have failed because of incompatible code. I've tested this against dLobatog/rails4 branch of foreman, and tests passed on ruby 2.0.0 and 2.1.5, I'll check 2.2 and 1.9.3 first thing tomorrow. (also, if tests do fail, I'll see how it looks on the CI, of course)

**Comment by johnpmitsch** [comment]
@unorthodoxgeek thanks. let us know if the changes aren't backwards compatible or if master cannot be updated for any reason, and we will switch to @komidore64 branch on jenkins.

**Comment by unorthodoxgeek** [comment]
ok, CI isn't running tests because there are a bunch of rubocop tests which seem to fail on master.

**Comment by dLobatog** [comment]
@unorthodoxgeek Mind to move attr_accessible to another PR, these can be merged now already, and they will fix some problems with master.

File: app/models/docker_registry.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class DockerRegistry < ActiveRecord::Base
   include Taxonomix
   include Encryptable
 
+  self.primary_key = 'id'
```
Comment (dLobatog) [inline app/models/docker_registry.rb]: Is this needed?

**Comment by unorthodoxgeek** [comment]
@dLobatog - opened #131 for attr_accessible

File: app/models/docker_registry.rb (hunk 1 above)
Comment (unorthodoxgeek) [inline app/models/docker_registry.rb]: I have no clue why I put it there... this should be the default, I'll try removing this

**Comment by unorthodoxgeek** [comment]
and, now it's also passing against develop. don't you love it when plugin migration is fully backward compatible with rails 3? :smile:

**Comment by johnpmitsch** [comment]
If this is still the PR that will include all of the rails 4 changes, can you add ```attr_accessible :email``` to app/models/foreman_docker/docker.rb?

**Comment by unorthodoxgeek** [comment]
let's merge this already.

**Comment by dLobatog** [comment]
Merged as 10020062b09907b4dbf51d709a82a6a8539263c1, thanks @unorthodoxgeek!

**Comment by domcleal** [comment]
This appears to have used the wrong ticket number, it "fixes" a Foreman core ticket so has closed #7230.  Please ensure you use a project-specific ticket in future.

**Pull request closed without merging**